### PR TITLE
Debugging via VSCode

### PR DESCRIPTION
**Pull request by kaermorchen**
I have added the debugging the extension through the `VS Code` tools.

TODO: 
- [x] debugging the extension through `VS Code`
- [x] debugging the server through `VS Code`
- [x] write/update DEBUGGING.md

### CODE REVIEW

**Comment by kaermorchen** [comment]
@lifeart I've saw you DEBUGGING.md, but it looks so difficult. I will try to find a more simple way.

**Comment by kaermorchen** [comment]
@lifeart I have done this PR. 

The task `Debugging the server through VS Code` is too complicated for me, too much breaking changes.

File: package.json [hunk 1]
```diff
@@ -113,7 +113,7 @@
     "watch": "tsc -watch -p ./",
     "postinstall": "yarn compile",
     "clean:cache": "yarn cache clean @lifeart/ember-language-server",
-    "build:langserver": "yarn --cwd ./node_modules/@lifeart/ember-language-server prepublish",
+    "build:langserver": "yarn --cwd ./node_modules/@lifeart/ember-language-server build:bundle:node",
```
Comment (lifeart) [inline package.json]: I think we don't need webpack bundled code for dev, we could use tsc artifacts

File: client.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// This work-around needs for working the VSCode's debugging
+try {
+  module.exports = require('./out/src/extension');
+}
+catch {
+  module.exports = require('./dist/node/client/nodeClientMain');
```
Comment (lifeart) [inline client.js:6]: in general, we need nodeClient only for prod mode, I don't think we need to follow try/catch here, in my understanding we could detect environment to choose actual server file..

Comment (lifeart) [reply client.js:6]: we resolve server path here: https://github.com/lifeart/vscode-ember/blob/master/src/extension.ts#L35
and this is how "prod" server file looks like: https://github.com/lifeart/vscode-ember/blob/master/start-server.js
but, for debugger it's better to use compiled version of https://github.com/lifeart/ember-language-server/blob/component-context-info-origin/src/start-server.ts

Comment (kaermorchen) [reply client.js:6]: > in general, we need nodeClient only for prod mode, I don't think we need to follow try/catch here, in my understanding we could detect environment to choose actual server file..

nodeClient needs for debugging extension in VSCode. After running `Launch UELS Extension` VSCode tries open `main` path (`./dist/node/client/nodeClientMain` in 'package.json') but output folder of tsc is `out` and VSCode crashes

**Comment by kaermorchen** [comment]
@lifeart How do you debug the extension without nodeClient?

File: client.js (hunk 2 above)
Comment (lifeart) [reply client.js:6]: @kaermorchen I'm pointing to '[start-server.ts](https://github.com/lifeart/ember-language-server/blob/component-context-info-origin/src/start-server.ts)' (compiled version) in linked `lifeart/ember-language-server` repo

`@lifeart/ember-language-server/lib/start-server.js`

File: .vscode/launch.json [hunk 3]
```diff
@@ -1,48 +1,28 @@
...
-			],
-			"stopOnEntry": false,
-			"sourceMaps": true,
-			"outFiles": [
-				"${workspaceRoot}/out/test/**/*.js"
-			],
-			"preLaunchTask": "npm"
-		},
-     {
-        "name":"Launch UELS Extension",
```
Comment (lifeart) [inline .vscode/launch.json:21]: let's keep `Launch UELS Extension` for future purposes

File: server.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// This work-around needs for working the VSCode's debugging
+try {
+  module.exports = require('@lifeart/ember-language-server/lib/server').default;
```
Comment (lifeart) [inline server.js:3]: I'm afraid we always have this path - https://www.runpkg.com/?@lifeart/ember-language-server@2.26.1/lib/index.js, so, there is no way to face "catch"

Comment (lifeart) [reply server.js:3]: https://npm-explorer.tk/?p=@lifeart/ember-language-server

Comment (lifeart) [reply server.js:3]: ah, found `.vscodeignore` config, looks like we a good here

**Comment by lifeart** [comment]
@kaermorchen thank you for contribution!

**Pull request merged**